tests/text/test_eed.py

```diff
@@ -0,0 +1,143 @@
+    to the examples defined in `tests.text.inputs` with the script from https://github.com/rwth-i6/ExtendedEditDistance.
+    """
+    HYPOTHESIS_A = "It is a guide to action which ensures that the military always obeys the commands of the party"
+    if HYPOTHESIS_A in preds:
+        return tensor(0.2425)
+    return tensor(0.1979)
+
+
+@pytest.mark.parametrize(
+    ["preds", "targets"],
+    [(_inputs_single_reference.preds, _inputs_single_reference.targets)],
+)
+class TestEED(TextTester):
+    @pytest.mark.parametrize("ddp", [False, True])
+    @pytest.mark.parametrize("dist_sync_on_step", [False, True])
+    def test_eed_class(self, ddp, dist_sync_on_step, preds, targets):
+        rwth_metric = partial(rwth_manual_metric)
+        self.run_class_metric_test(
+            ddp=ddp,
+            preds=preds,
+            targets=targets,
+            metric_class=EED,
+            sk_metric=rwth_metric,
+            dist_sync_on_step=dist_sync_on_step,
+            input_order=INPUT_ORDER.TARGETS_FIRST,
+        )
+
+    def test_eed_functional(self, preds, targets):
+        rwth_metric = partial(rwth_manual_metric)
+        self.run_functional_metric_test(
+            preds,
+            targets,
+            metric_functional=eed,
+            sk_metric=rwth_metric,
+            input_order=INPUT_ORDER.TARGETS_FIRST,
+        )
+
+    def test_eed_differentiability(self, preds, targets):
+        self.run_differentiability_test(
+            preds=preds,
+            targets=targets,
+            metric_module=EED,
+            metric_functional=eed,
+            input_order=INPUT_ORDER.TARGETS_FIRST,
+        )
+
+
+# test blank edge cases
+def test_eed_empty_functional():
+    ref = [[]]
+    hyp = []
+    assert eed(ref, hyp) == tensor(0.0)
+
+
+def test_eed_empty_class():
+    eed_metric = EED()
+    ref = [[]]
+    hyp = []
+    assert eed_metric(ref, hyp) == tensor(0.0)
+
+
+def test_eed_empty_with_non_empty_hyp_functional():
+    ref = [[]]
+    hyp = ["python"]
+    assert eed(ref, hyp) == tensor(0.0)
+
+
+def test_eed_empty_with_non_empty_hyp_class():
+    eed_metric = EED()
+    ref = [[]]
+    hyp = ["python"]
+    assert eed_metric(ref, hyp) == tensor(0.0)
+
+
+def test_eed_return_sentence_level_score_functional():
+    ref = _inputs_single_sentence_multiple_references.targets
+    hyp = _inputs_single_sentence_multiple_references.preds
+    _, sentence_eed = eed(ref, hyp, return_sentence_level_score=True)
+    isinstance(sentence_eed, Tensor)
+
+
+def test_eed_return_sentence_level_class():
+    metric = EED(return_sentence_level_score=True)
+    ref = _inputs_single_sentence_multiple_references.targets
+    hyp = _inputs_single_sentence_multiple_references.preds
+    _, sentence_eed = metric(ref, hyp)
+    isinstance(sentence_eed, Tensor)
+
+
+# test parallel vs sequential computations
+def test_parallelisation_eed():
+    references = _inputs_multiple_references.targets
+    hypotheses = _inputs_multiple_references.preds
```

Comment:
```suggestion
    references = _inputs_multiple_references.targets[0]
    hypotheses = _inputs_multiple_references.preds[0]
```

It's sufficient to run this test with a single batch of two examples. With the previous setting with two batches, you actually pass, e.g. for hypothesis, inputs in the form of `Tuple[Tuple[str, str], Tuple[str, str]]` which results in the error as it cannot be well processed.

Reply:
got it